Document that `gh run download` downloads the latest artifact by default

sato11 opened the pull request:
This commit addresses the documentation issue.

The discussion at #7018 has confirmed that the current behavior of `gh run download -n <name>` downloads the latest artifact.

Although the behavior is not intentional, it is the one that I wish should be documented and the future releases should warn before a breaking change.

---

This is associated with https://github.com/cli/cli/issues/7018#issuecomment-1488648336 but I guess this does not fully resolve the whole issue. Therefore I'm making a reference without the intention to close it. Let me know if you need anything else. 

<!--
  Thank you for contributing to GitHub CLI!
  To reference an open issue, please write this in your description: `Fixes #NUMBER`
-->

andyfeller commented on pkg/cmd/run/download/download.go:
```diff
@@ -57,7 +57,10 @@ func NewCmdDownload(f *cmdutil.Factory, runF func(*DownloadOptions) error) *cobr
 		  $ gh run download <run-id> -n <name>
 
-		  # Download specific artifacts across all runs in a repository
+		  # Download the latest artifact with a specific name
+		  $ gh run download -n <name>
+
+		  # Download multiple latest artifacts by name
 		  $ gh run download -n <name1> -n <name2>
```
@sato11 : Instead of changing the explanation here within the examples, **perhaps we should add more content to the `Long` section that explains more fully?**

```
		Long: heredoc.Doc(`
			Download artifacts generated by a GitHub Actions workflow run.

			The contents of each artifact will be extracted under separate directories based on
			the artifact name. If only a single artifact is specified, it will be extracted into
			the current directory.

			By default, this command downloads the latest artifact created and uploaded through
			GitHub Actions. Because workflows can deleted or overwrite artifacts, `<run-id>`
			must be used to select an artifact from a specific workflow run.
		`),
```

sato11 replied:
It sounds great to me and I will be undoing the current changes and opting for your idea. Thanks for the suggestion 🙂

sato11 replied:
Made the change, rebased and renamed the title. Will you take a look again? @andyfeller

andyfeller reviewed (changes requested):
@sato11 : Thank you for helping to make `gh` incrementally better! 🙇

Minor suggestion on the documentation approach.

sato11 committed:
Document that `gh run download` downloads the latest artifact by default

This commit addresses the documentation issue.

The discussion at #7018 has confirmed that it is undocumented that
the current behavior of `gh run download` with `-n` and no `run-id`
downloads the latest artifact.

Although the behavior has not been created intentionally, it is the one
that should be documented and the future releases should warn before
a breaking change.

andyfeller reviewed (approved):
🙇 thanks @sato11! :shipit:
